examples/hubert/dataset/hubert_dataset.py

```diff
@@ -0,0 +1,317 @@
+            bucket_id = bucket_ids[i]
+            if bucket_id in buckets:
+                buckets[bucket_id].append(i)
+            else:
+                buckets[bucket_id] = [i]
+        for k in buckets:
+            random.shuffle(buckets[k])
+            buckets[k] = torch.as_tensor(buckets[k], dtype=torch.int)
+        return buckets
+
+    def __iter__(self) -> Iterator[List[int]]:
+        iter_list = []
+        total_len = 0
+        batch = []
+        len_list = self.data_source.len_list
+        if self.max_token_count:
+            for k in self.buckets.keys():
+                for i in range(self.buckets[k].size(0)):
+                    index = self.buckets[k][i]
+                    if total_len + len_list[index] > self.max_token_count:
+                        iter_list.append(batch)
+                        batch = [index]
+                        total_len = len_list[index]
+                    else:
+                        batch.append(index)
+                        total_len += len_list[index]
+        else:
+            for i in range(self.buckets[k].size(0)):
+                index = self.buckets[k][i]
+                if len(batch) == self.batch_size:
+                    iter_list.append(batch)
+                    batch = [index]
+                    total_len = len_list[index]
+                else:
+                    batch.append(index)
+                    total_len += len_list[index]
+
+        for batch in iter_list:
+            yield batch
+
+    def __len__(self):
+        return len(self.data_source)
+
+
+class HuBERTDataSet(Dataset):
+    """Create a Dataset for HuBERT model training and fine-tuning.
+
+    Args:
+        exp_dir (str or Path): The root directory of the ``.tsv`` file list.
+        dataset (str): The dataset for training. Options: [``librispeech``, ``librilight``].
+        phase_type (str): The type of the dataset. Options: [``train``, ``valid``].
+        min_sample (int): The minimum number of audio samples in the dataset. (Default: 32000)
+        max_sample (int): The maximum number of audio samples in the dataset. (Default: 250000)
+    """
+    def __init__(
+        self,
+        exp_dir: Union[str, Path],
+        dataset: str,
+        phase_type: str,
+        min_sample: int = 32000,
+        max_sample: int = 250000,
+    ) -> None:
+        self.exp_dir = Path(exp_dir)
+        tsv_dir = self.exp_dir / "tsv"
+        label_dir = self.exp_dir / "label"
+        f_list, ind_list, len_list = self._get_lists(
+            tsv_dir,
+            dataset,
+            phase_type,
+            min_sample,
+            max_sample
+        )
+        self.f_list, self.ind_list, self.len_list = f_list, ind_list, len_list
+        self.labels = self._load_labels(label_dir, dataset, phase_type)
+
+    def __len__(self):
+        return len(self.f_list)
+
+    def _get_lists(
+        self,
+        tsv_dir: Path,
+        dataset: str,
+        phase_type: str,
```

Comment:
would `subset` be a better name for this?

Reply:
Sounds good. Let me replace with it.